### ISSUE TITLE
Question regarding compatability with Mavic Mini

### ISSUE THREAD

**Issue by 46cv8** [Toemsel/dji-wifi-tools#4]
Hey,

I'm sorry to ask a question here as it isn't an issue with your code, but I couldn't find any other way to reach you.
In the readme you list compatability with Mavic Pro 1, Phantom 3 and Phantom 4.
I'm curious if you think this would be compatible with the Mavic Mini too.
You cannot connect directly to the drone using a phone, but need to connect the drone through a controller.
However unlike the mavic mini 2 the mavic mini 1 does not use occusync.
I've included some of the details below regarding the wifi config on the controller and the drone.
I'm curious with how this compares with the wifi config you discovered on your drone as described on your blog.
https://www.indie-dev.at/2021/03/15/day-1-getting-started/
Do you think a similar approach to connecting to the AC from a PC, then exposing an AP for the controller to connect to might work here too?
I am a little uncertain as it looks like there is some kind of paired password stored, and it's unclear to me if I would need to do something additional to facilitate the drone/controller pairing process or not.

```
(extracted from controller via serial over ethernet)
=====================
RC wifi.conf
=====================
mac1=XX:XX:XX:XX:XX:X3
mac2=XX:XX:XX:XX:XX:X4
rc_ssid=Spark-RC-XXXXXX
rc_passwd=12341234
phone_5gcap=1
country=ZZ
has_set_mac=XX:XX:XX:XX:XX:X3
update_ok=0
update_src=10
11p_freq=2432
11p_mac=YY:YY:YY:YY:YY:Y2
sky_passwd=ZZZZZZZZ
11p_flag=1
11p_mode=1
sky_ssid=WM160-#####
disable_country=1
```

(dumped from firmware)
```
=====================
$CONTROL_DIR/hostapd.conf
=====================
driver=nl80211
interface=wlan0
ctrl_interface=/tmp/hostapd
ssid=WM160-5g_test
country_code=US
ieee80211n=1
hw_mode=a
channel=157
max_num_sta=1
ap_max_inactivity=10
beacon_int=200
ignore_broadcast_ssid=0
wpa=2
wpa_key_mgmt=WPA-PSK
wpa_pairwise=CCMP
wpa_passphrase=12341234
=====================
athsoftap.sh
=====================
CHIP_SN=$RANDOM
AP_SSID="WM160-"${CHIP_SN}
# WIFI hostapd.conf
if [ ! -f "$AP_CONF" ]; then
echo "driver=nl80211"                         >> $AP_CONF
echo "interface=wlan0"                        >> $AP_CONF
echo "ctrl_interface=/tmp/hostapd"            >> $AP_CONF
echo "ssid="${AP_SSID}                        >> $AP_CONF
echo "country_code=CN"                        >> $AP_CONF
echo "ieee80211n=1"                           >> $AP_CONF
echo "ap_max_inactivity=10"                   >> $AP_CONF
echo "hw_mode=a"                              >> $AP_CONF
echo "channel=153"                            >> $AP_CONF
echo "ignore_broadcast_ssid=0"                >> $AP_CONF
echo "wpa=2"                                  >> $AP_CONF
echo "wpa_key_mgmt=WPA-PSK"                   >> $AP_CONF
echo "wpa_pairwise=CCMP"                      >> $AP_CONF
echo "wpa_passphrase=12341234"                >> $AP_CONF
f
```

You did some really great work, I'de love to try and leverage it, to perhaps even control the drone via a laptop :)

**Comment by 46cv8** [comment]
Thanks for the reply! The AC's AP is hidden I believe. Based on what I can tell from configurations and my frequency scanner it is indeed using 2.4 Ghz just that the access point is hidden. The AP name and password are based on the device serial and should be detectable during the connection process (although I haven't been successful in forming it) which is how I think the DJI aeroscope may be able to track drones back to whoever activated/purchased them.

I will let you know if I am successful in establishing a connection. I'm a little concerned it may not be a regular wifi connection but the scripts on the remote seem to indicate it is.

**Comment by kripper** [comment]
@46cv8 any luck?

**Comment by 46cv8** [comment]
> @46cv8 any luck?

No but I stopped part way. Basically the core issue is that mavic mini using some special 5mhz quarter bandwidth mode of wifi. Only certain wifi cards support it. On the dji-rev chat there was info regarding which wifi cards. I bought two wifi cards but I ran out of time to test it as the laptop I was going to put the cards in stopped working. The cards that support the are ones like the following.

Atheros ath9k Atheros QCNFA222 AR5BWB222
https://www.aliexpress.com/item/33011203971.html

Here's the kismet config for the 5mhz channels
```
/usr/local/etc/kismet_site.conf:
source=wlan10k0:add_channels="149W5,153W5,157W5,161W5,165W5,151W5,155W5,159W5,163W5,153W5,1W5,2W5,3W5,4W5,5W5,6W5,7W5,8W5,9W5,10W5,11W5,12W5,13W5,14W5"
```
